src/api/controllers/voteController.ts

```diff
@@ -0,0 +1,62 @@
+import { BaseAPI } from "../abstract";
+import {
+  CreateVoteResponse,
+  CreateVoteRequest,
+  PartialVoteRequest,
+} from "./types/voteController.type";
+
+export class VoteController extends BaseAPI {
+  private baseUrl = "https://api.thecatapi.com/v1";
+
+  private headers() {
+    if (!process.env.API_KEY) {
+      throw new Error("API_KEY is not defined");
+    }
+    return {
+      "x-api-key": process.env.API_KEY,
+      "Content-Type": "application/json",
+    };
+  }
+
+  async addVote(
+    data: CreateVoteRequest | PartialVoteRequest
+  ): Promise<CreateVoteResponse> {
+    const response = await this.request.post(`${this.baseUrl}/votes`, {
+      data,
+      headers: this.headers(),
+    });
+    if (!response.ok()) {
+      const errorMessage = await response.text();
+      throw new Error(`${errorMessage}`);
+    }
+    return (await response.json()) as CreateVoteResponse;
```

Comment:
```suggestion
    return (await response.json());
```

here and at the other methods, no need to cast the type, since the `response.json()` returns `any` and the method  (`addVote`) return type is specified.

Reply:
Fixed